js/directives/lucca/user-picker.js

```diff
@@ -0,0 +1,322 @@
+										user.hasHomonyms = true;
+										otherUser.hasHomonyms = true;
+										if (!hasHomonyms) {
+											hasHomonyms = true;
+										}
+									}
+								});
+
+								// Search homonyms in selected users (used by UserPickerMultiple only)
+								_.each($scope.selected.users, function (selectedUser) {
+									if ((user.firstName === selectedUser.firstName) && (user.lastName === selectedUser.lastName)) {
+										user.hasHomonyms = true;
+										selectedUser.hasHomonyms = true;
+										if (!hasHomonyms) {
+											hasHomonyms = true;
+										}
+									}
+								});
+							});
+			return hasHomonyms;
+		};
+
+		var getHomonymsProperties = function () {
+			var homonyms = _.where($scope.users, { hasHomonyms: true });
+			var homonymsArray = [];
+
+			loadPropertiesFromUrls(homonyms).then(function (response) {
+								// Add homonyms properties for each user
+								_.each(homonyms, function (user, item) {
+									// Get the returned user
+									var userWithProps = response[item].data.data.items[0];
+
+									// Add each property to the user
+									_.each($scope.properties, function (prop) {
+										var newProp = prop.split('.')[0];
+										user[newProp] = userWithProps[newProp];
+									});
+								});
+							});
+		};
+
+		var loadPropertiesFromUrls = function (homonyms) {
+			var urlCalls = [];
+			var queries = [];
+			var templateQuery = "/api/v3/users?id=";
+			var fields = "&fields=id,name,firstname,lastname,dtcontractend";
+			var deferred = $q.defer();
+
+			// WARNING: Do not check if the properties exist!
+			// WARNING: If they do not exist, the request will fail
+			_.each($scope.properties, function (prop) {
+				fields += "," + prop;
+			});
+
+			_.each(homonyms, function (user) {
+				queries.push(templateQuery + user.id + fields);
+			});
+
+			_.each(queries, function (query) {
+				urlCalls.push($http.get(query));
+			});
+
+			$q.all(urlCalls)
+			.then(
+				function (response) {
+					deferred.resolve(response);
+				},
+				function (error) {
+					deferred.reject(error);
+				}
+				);
+			return deferred.promise;
+		};
+
+		var countTimeout = function (input, delay) {
+			delay = delay || 5000; // default delay value 1s
+
+			// cancel previous timeout
+			if (timeout.count) {
+				console.log("-- cancel previous timeout");
+				$timeout.cancel(timeout.count);
+			}
+
+			// launch new timeout to save in _delay_ 
+			timeout.count = $timeout(function () {
+				doCount(input);
+			}, delay);
+		};
+
+		var doCount = function (input) {
+			var query = "/api/v3/users?name=like," + input + "&fields=collection.count"; // query for count
+
+			delete timeout.count;
+
+			$http.get(query)
+			.then(function (response) {
+					$scope.count = response.data.data.count; // To update with right query
+					_.last($scope.users).overflow = MAX_COUNT + "/" + $scope.count;
+				});
+		};
+
+		$scope.getProperty = function (user, prop) {
```

Comment:
the five last methods are in order
 - `var getHomonymsProperties = function (){...};`
 - `var loadPropertiesFromUrls = function (homonyms) {...}`
 - `var countTimeout = function (input, delay) {...}`
 - `var doCount = function (input) {...}`
 - `$scope.getProperty = function (user, prop) {...}`

you should group the different method by what feature they implement. here you have 2 methods to handle pagination in the middle of 3 method that handle the homonyms and it's a bit jarring